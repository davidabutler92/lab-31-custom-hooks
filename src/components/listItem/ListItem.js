import React from 'react';
import PropTypes from 'prop-types';

function ListItem({ name, image }) {
  return (
    <li>
      <img src={image} alt={name} />
      <h2>{name}</h2>
    </li>
  );
}

ListItem.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
};

export default ListItem;
