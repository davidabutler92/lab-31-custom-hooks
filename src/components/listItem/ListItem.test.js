import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListItem from './ListItem';

describe('DetailsPage component', () => {
  afterEach(() => cleanup());
  it('renders a list item', () => {
    const { asFragment } = render(
      <ListItem name="some name" image="some image" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
