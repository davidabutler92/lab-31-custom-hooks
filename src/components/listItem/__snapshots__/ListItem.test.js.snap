// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component renders a list item 1`] = `
<DocumentFragment>
  <img
    alt="some name"
    src="some image"
  />
  <h2>
    some name
  </h2>
</DocumentFragment>
`;
